### PR TITLE
Moving tokenization logic to DbotPreprocessTextData

### PR DESCRIPTION
**Pull request by lperryp**
## Description
Replacing WordTokenizerV2 call to ApiModule 

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ApiModules/Scripts/MLApiModule/MLApiModule.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+import string
+from html.parser import HTMLParser
+from html import unescape
+from re import compile as _Re
+import pandas as pd
+
+
+def handle_long_text(t, input_length):
+    if input_length == 1:
+        return_error("Input text length ({}) exceeds the legal maximum length for preprocessing".format(len(t)))
```
Comment (erezh31) [inline Packs/ApiModules/Scripts/MLApiModule/MLApiModule.py:13]: why we return error in that case?
I think return error do sys.exit, do we want that?

File: Packs/Base/Scripts/DBotPreprocessTextData/DBotPreprocessTextData.yml [hunk 2]
```diff
@@ -103,7 +107,8 @@ args:
 - auto: PREDEFINED
   default: false
   defaultValue: English
-  description: The language of the input text. Default is "English". Can be "English", "German", "French", "Spanish", "Portuguese", "Italian", "Dutch", or "Other".
+  description: The language of the input text. Default is "English". Can be "English",
+    "German", "French", "Spanish", "Portuguese", "Italian", "Dutch", or "Other".
```
Comment (erezh31) [inline Packs/Base/Scripts/DBotPreprocessTextData/DBotPreprocessTextData.yml:111]: I think we need to update the languages, no?

File: Packs/ApiModules/Scripts/MLApiModule/TestData/input_json_file_test [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[{"body": "This is a sample of the body", "subject": "This is a sample of the subject"},
```
Comment (erezh31) [inline Packs/ApiModules/Scripts/MLApiModule/TestData/input_json_file_test:1]: let's have some UTF-8 chars in the file

File: Packs/ApiModules/Scripts/MLApiModule/TestData/input_json_file_test [hunk 4]
```diff
@@ -0,0 +1,2 @@
...
+{"body2": "This is a sample of the body2", "subject2": "This is a sample of the subject2", "subject": "This is a sample of the subject"}]
```
Comment (erezh31) [inline Packs/ApiModules/Scripts/MLApiModule/TestData/input_json_file_test:2]: maybe add also a CSV - we use it also in the train from file. Also include utf8 characters in the CSV

File: Packs/ApiModules/Scripts/MLApiModule/MLApiModule.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        else:
+            res = demisto.getFilePath(input_data)
+            if not res:
+                return_error("Entry {} not found".format(input_data))
+            file_path = res['path']
+            if input_type.startswith('json'):
+                with open(file_path, 'r') as f:
+                    file_content = f.read()
+        if input_type.startswith('csv'):
+            return pd.read_csv(file_path).fillna('').to_dict(orient='records')
```
Comment (erezh31) [inline Packs/ApiModules/Scripts/MLApiModule/MLApiModule.py:231]: add test to csv case as mentioned below

File: Packs/ApiModules/Scripts/MLApiModule/MLApiModule.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+        return ' '.join(tokens_list).strip(), ' '.join(hashed_tokens_list) if len(
+            hashed_tokens_list) > 0 else None, original_words_to_tokens, hashed_words_to_tokens
+
+    def tokenize_text_other(self, text):
+        tokens_list = []
+        tokenization_method = self.tokenization_method
+        if tokenization_method == 'byWords':
+            original_words_to_tokens = {}
+            for t in text.split():
+                token_without_punct = ''.join([c for c in t if c not in string.punctuation])
```
Comment (erezh31) [inline Packs/ApiModules/Scripts/MLApiModule/MLApiModule.py:121]: I think NLTK has punct tokenizer, maybe we can consider using that

**Review by erezh31 — CHANGES_REQUESTED** [review]

**Comment by kirbles19** [comment]
@lperryp Done

**Comment by lperryp** [comment]
Hard to follow changes, will re-open and apply changes in new pr: https://github.com/demisto/content/pull/9594
